### PR TITLE
Ginkgo: Added Envoy on ginkgo test

### PR DESCRIPTION
**Pull request by eloycoto**
- Added Envoy options on the PR tests

Signed-off-by: Eloy Coto <eloy.coto@gmail.com>

### CODE REVIEW

**Comment by eloycoto** [comment]
test-all-ginkgo

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-all-ginkgo

**Comment by eloycoto** [comment]
test-all-ginkgo

**Comment by eloycoto** [comment]
test-all-ginkgo

**Comment by eloycoto** [comment]
test-all-ginkgo

**Comment by eloycoto** [comment]
test-me-please

File: test/Vagrantfile [hunk 1]
```diff
@@ -5,7 +5,7 @@ $BUILD_NUMBER = ENV['BUILD_NUMBER'] || "0"
 $JOB_NAME = ENV['JOB_BASE_NAME'] || "LOCAL"
 $K8S_VERSION = ENV['K8S_VERSION'] || "1.7"
 $K8S_NODES = (ENV['K8S_NODES'] || "2").to_i
-
+$CILIUM_USE_ENVOY = (ENV['CILIUM_USE_ENVOY'] || "0")
```
Comment (jrajahalme) [inline test/Vagrantfile]: Defining CILIUM_USE_ENVOY as "0" does not have the desired effect as envoy/Makefile tests if CILIUM_USE_ENVOY is defined and ignores the value.

Also, to reflect the main Vagrantfile, IMO test/Vagrantfile should test or even read the CILIUM_USE_ENVOY environment variable, but unconditionally define CILIUM_USE_ENVOY=1 for the provisioning scripts.

Once we have everything successfully moved over to Envoy, we'll be removing "CILIUM_USE_ENVOY" env variable altogether.

**Review by jrajahalme — CHANGES_REQUESTED** [review]

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-all-ginkgo

**Review by ianvernon — APPROVED** [review]

**Comment by ianvernon** [comment]
LGTM, @jrajahalme can you confirm that the output from Bazel looks OK?:
```
16:08:52 ==> runtime: make[1]: Entering directory '/go/src/github.com/cilium/cilium/envoy'
16:08:52 ==> runtime: bazel --batch build --spawn_strategy=standalone --genrule_strategy=standalone -c opt //:envoy
16:08:55 ==> runtime: Loading: 
16:08:55 ==> runtime: Loading: 0 packages loaded
16:08:56 ==> runtime: Analyzing: target //:envoy (5 packages loaded)
16:08:56 ==> runtime: External dependency cache directory /home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f
16:08:56 ==> runtime: make[2]: Entering directory '/home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy_deps'
16:08:56 ==> runtime: make[2]: '/home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/cares.dep' is up to date.
16:08:56 ==> runtime: make[2]: '/home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/libevent.dep' is up to date.
16:08:56 ==> runtime: make[2]: '/home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/gperftools.dep' is up to date.
16:08:56 ==> runtime: make[2]: '/home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/luajit.dep' is up to date.
16:08:56 ==> runtime: make[2]: '/home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/nghttp2.dep' is up to date.
16:08:56 ==> runtime: make[2]: '/home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/boringssl.dep' is up to date.
16:08:56 ==> runtime: make[2]: '/home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/yaml-cpp.dep' is up to date.
16:08:56 ==> runtime: make[2]: '/home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/zlib.dep' is up to date.
16:08:56 ==> runtime: make[2]: Leaving directory '/home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy_deps'
16:08:56 ==> runtime: 
16:08:56 ==> runtime: real	0m0.051s
16:08:56 ==> runtime: user	0m0.000s
16:08:56 ==> runtime: sys	0m0.000s
16:08:56 ==> runtime: DEBUG: /home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy/bazel/repositories.bzl:76:5: Fetching external dependencies...
16:08:56 ==> runtime: DEBUG: /home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy/bazel/repositories.bzl:82:5: 
16:08:56 ==> runtime: DEBUG: /home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy/bazel/repositories.bzl:83:5: External dependency cache directory /home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f
16:08:56 ==> runtime: make[2]: Entering directory '/home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy_deps'
16:08:56 ==> runtime: make[2]: '/home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/cares.dep' is up to date.
16:08:56 ==> runtime: make[2]: '/home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/libevent.dep' is up to date.
16:08:56 ==> runtime: make[2]: '/home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/gperftools.dep' is up to date.
16:08:56 ==> runtime: make[2]: '/home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/luajit.dep' is up to date.
16:08:56 ==> runtime: make[2]: '/home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/nghttp2.dep' is up to date.
16:08:56 ==> runtime: make[2]: '/home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/boringssl.dep' is up to date.
16:08:56 ==> runtime: make[2]: '/home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/yaml-cpp.dep' is up to date.
16:08:56 ==> runtime: make[2]: '/home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy_deps_cache_2c744dffd279d7e9e0910ce594eb4f4f/zlib.dep' is up to date.
16:08:56 ==> runtime: make[2]: Leaving directory '/home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy_deps'
16:08:56 ==> runtime: 
16:08:56 ==> runtime: real	0m0.051s
16:08:56 ==> runtime: user	0m0.000s
16:08:56 ==> runtime: sys	0m0.000s
16:08:56 ==> runtime: DEBUG: /home/vagrant/.cache/bazel/_bazel_vagrant/e3bb2978717df7858b43846d8c1242eb/external/envoy/bazel/repositories.bzl:84:5: External dep build exited with return code: 0
16:08:57 ==> runtime: Analyzing: target //:envoy (52 packages loaded)
16:08:57 ==> runtime: INFO: Analysed target //:envoy (105 packages loaded).
16:08:57 ==> runtime: INFO: Found 1 target...
16:08:57 ==> runtime: [0 / 1] BazelWorkspaceStatusAction stable-status.txt
16:09:00 ==> runtime: [517 / 584] no action running
16:09:02 ==> runtime: [749 / 905] no action running
16:09:06 ==> runtime: [896 / 905] no action running
16:09:08 ==> runtime: [907 / 909] Compiling cilium_l7policy.cc; 1s local
16:09:15 ==> runtime: [908 / 909] Linking envoy
16:09:19 ==> runtime: [908 / 909] Linking envoy; 5s local
16:09:24 ==> runtime: [908 / 909] Linking envoy; 11s local
16:09:27 ==> runtime: Target //:envoy up-to-date:
16:09:27 ==> runtime:   bazel-bin/envoy
16:09:27 ==> runtime: INFO: Elapsed time: 32.933s, Critical Path: 21.56s
16:09:27 ==> runtime: INFO: Build completed successfully, 5 total actions
16:09:27 ==> runtime: bazel-out/host/bin/external/com_google_protobuf/protoc --proto_path=. -I bazel-envoy/external/com_google_protobuf/src -I bazel-envoy/external/googleapis -I bazel-envoy/external/com_lyft_protoc_gen_validate --go_out=plugins=grpc:../pkg/envoy ./*.proto
```

**Comment by jrajahalme** [comment]
> On Jan 23, 2018, at 9:26 AM, Ian Vernon <notifications@github.com> wrote:
> 
> LGTM, @jrajahalme <https://github.com/jrajahalme> can you confirm that the output from Bazel looks OK?:
> 
> 

Yes it does!

  Jarno

File: test/Vagrantfile [hunk 2]
```diff
@@ -5,7 +5,7 @@ $BUILD_NUMBER = ENV['BUILD_NUMBER'] || "0"
 $JOB_NAME = ENV['JOB_BASE_NAME'] || "LOCAL"
 $K8S_VERSION = ENV['K8S_VERSION'] || "1.7"
 $K8S_NODES = (ENV['K8S_NODES'] || "2").to_i
-$CILIUM_USE_ENVOY = (ENV['CILIUM_USE_ENVOY'] || "0")
+$CILIUM_USE_ENVOY = (ENV['CILIUM_USE_ENVOY'] || "1")
```
Comment (jrajahalme) [inline test/Vagrantfile]: I'd rather not define $CILIUM_USE_ENVOY in the Vagrantfile at all.

File: test/Vagrantfile [hunk 3]
```diff
@@ -32,6 +32,7 @@ Vagrant.configure("2") do |config|
         server.vm.provision "file", source: "provision", destination: "/tmp/provision"
         server.vm.provision "shell" do |sh|
             sh.path = "./provision/runtime_install.sh"
+            sh.env = {"CILIUM_USE_ENVOY" => "#{$CILIUM_USE_ENVOY}"}
```
Comment (jrajahalme) [inline test/Vagrantfile]: We don't need the $CILIUM_USE_ENVOY variable in the Vagrantfile, so maybe do it like this:
```sh.env = {"CILIUM_USE_ENVOY" => "1"}```

**Review by jrajahalme — CHANGES_REQUESTED** [review]

File: test/Vagrantfile (hunk 3 above)
Comment (tgraf) [reply test/Vagrantfile]: Can we just get rid of `CILIUM_USE_ENVOY`?

**Comment by eloycoto** [comment]
test-me-please

**Comment by eloycoto** [comment]
test-all-ginkgo

**Review by ianvernon — APPROVED** [review]

**Comment by jrajahalme** [comment]
> On Jan 23, 2018, at 6:34 PM, Thomas Graf <notifications@github.com> wrote:
> 
> @tgraf commented on this pull request.
> 
> In test/Vagrantfile <https://github.com/cilium/cilium/pull/2598#discussion_r163439935>:
> 
> > @@ -32,6 +32,7 @@ Vagrant.configure("2") do |config|
>          server.vm.provision "file", source: "provision", destination: "/tmp/provision"
>          server.vm.provision "shell" do |sh|
>              sh.path = "./provision/runtime_install.sh"
> +            sh.env = {"CILIUM_USE_ENVOY" => "#{$CILIUM_USE_ENVOY}"}
> Can we just get rid of CILIUM_USE_ENVOY?
> 
I guess at this point we can. I’ll prepare a PR, but let’s not delay this PR for that.

File: test/Vagrantfile [hunk 4]
```diff
@@ -32,6 +31,7 @@ Vagrant.configure("2") do |config|
         server.vm.provision "file", source: "provision", destination: "/tmp/provision"
         server.vm.provision "shell" do |sh|
             sh.path = "./provision/runtime_install.sh"
+            sh.env = {}
```
Comment (jrajahalme) [inline test/Vagrantfile:34]: Not sure if it is meaningful to have empty env like this.

File: test/provision/compile.sh [hunk 5]
```diff
@@ -25,7 +26,7 @@ then
         echo "Not on master K8S node; no need to compile Cilium container"
     fi
 else
-    make
+    sudo -u vagrant -H -E make CILIUM_USE_ENVOY=1 PKG_BUILD=1
     make install
```
Comment (jrajahalme) [inline test/provision/compile.sh:30]: make will skip install of envoy unless you add CILIUM_USE_ENVOY=1 also here.

File: test/provision/compile.sh [hunk 6]
```diff
@@ -25,7 +26,7 @@ then
         echo "Not on master K8S node; no need to compile Cilium container"
     fi
 else
-    make
+    sudo -u vagrant -H -E make CILIUM_USE_ENVOY=1 PKG_BUILD=1
```
Comment (jrajahalme) [inline test/provision/compile.sh:29]: I have not seen this pattern before (sudo -u vagrant). Do we need it here to make the build run by user "vagrant"?

Also, PKG_BUILD=1 is so far only used for container builds, where the build is done as root. envoy/Makefile will build the release version if PKG_BUILD is defined. Not sure we want it like that. This should work when not in a container:

```make CILIUM_USE_ENVOY=1```

Comment (jrajahalme) [reply test/provision/compile.sh:29]: If the build is done as root you should add ```PKG_BUILD=1```

File: test/Vagrantfile (hunk 4 above)
Comment (jrajahalme) [reply test/Vagrantfile:34]: For now CILIUM_USE_ENVOY=1 needs to be defined in the runtime environment of cilium-agent. It seems to me that this is the place where it needs to be defined.

**Review by jrajahalme — COMMENTED** [review]
Sorry for the poorly documented use of CILIUM_USE_ENVOY. As of now it seems to me that this PR would not run Envoy, but use the golang proxy instead.

**Comment by tgraf** [comment]
@jrajahalme Sorry, we had some miscommunication. @eloycoto mentioned on slack that all your feedback was already addressed. Please sync up with @eloycoto to fix up whatever is required.

**Comment by jrajahalme** [comment]
I’ll post a PR to remove CILIUM_USE_ENVOY soon, I’ll fix this up as well.

  Jarno

**Comment by jrajahalme** [comment]
#2625

**Pull request merged**